### ISSUE TITLE
num_adaptation_steps is not 1 error.

### ISSUE THREAD

**Issue by frankchen121212** [tristandeleu/pytorch-maml#13]
Hi,

Thank you for the nice code shared. I've encuntered with the problem when setting the num_adaptation_steps not to be 1 (adaptation step = 10 for example)and in the function "adapt" of the class "ModelAgnosticMetaLearning" , the following error occured. I'm wondering if you have the same issure before ? 

![image](https://user-images.githubusercontent.com/22726349/95845859-484cd500-0d7d-11eb-8bdb-e56eddce8c79.png)

Since I'm utilizing meta learning regarding to a different classification problem, the code would be slightly different.
I think the error occured in the loop:

for step in range(num_adaptation_steps):
            logits = self.model(inputs, params=params)
            inner_loss = self.loss_function(logits, targets)
            results['inner_losses'][step] = inner_loss.item()

            if (step == 0) and is_classification_task:
                results['accuracy_before'] = compute_accuracy(logits, targets)

            self.model.zero_grad()
            params = gradient_update_parameters(self.model, inner_loss,
                step_size=step_size, params=params,
                first_order=(not self.model.training) or first_order)

        return params, results

I haven't change the funciton but feeding the loop with different type of data. If you have time, would you please look into that?
Thank you!

**Comment by frankchen121212** [comment]
However, when setting the num_adaptation_steps at 1, it runs perfectly fine.

**Comment by tristandeleu** [comment]
Thank you for the kind words! I have not had any issue with `num_adaptation_steps > 1` in the past. Can you attach the code of the model you are using?

**Comment by frankchen121212** [comment]
Ohhh, Thank you for the reminding me that there could be an error in the model, I though there was something wrong with the loop in the function. But ,actually I just found an error in my model. Now it runs perfectly fine !

Thank you again for the nice code! I'll close the issue.  Have a nice day !
